### PR TITLE
Allow to set logger of `LoggingDecoratorBuilder` using name

### PR DESCRIPTION
**Pull request by ghkim3221**
### Motivation

#3029 

### Modification

Add `loggerName(String)` to `LoggingDecoratorBuilder`.

### Result

Users can configure `LoggingDecoratorBuilder`'s `logger` only using name.
Prevent passing null to `LoggingDecoratorBuilder.logger()` because of compile failure (have to case null as `Logger` of `String`).

### CODE REVIEW

File: core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java [hunk 1]
```diff
@@ -82,6 +83,16 @@ public LoggingDecoratorBuilder logger(Logger logger) {
         return this;
     }
 
+    /**
+     * Sets ths {@link Logger} to use when logging using loggerName.
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java:87]: Sets the specified `loggerName` which is used to create a new {@link Logger}.

Comment (ghkim3221) [reply core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java]: Thanks 😄

File: core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java [hunk 2]
```diff
@@ -82,6 +83,16 @@ public LoggingDecoratorBuilder logger(Logger logger) {
         return this;
     }
 
+    /**
+     * Sets the specified loggerName which is used to create a new {@link Logger}.
+     * If unset, a default {@link Logger} will be used.
+     */
+    public LoggingDecoratorBuilder logger(String loggerName) {
+        requireNonNull(loggerName, "loggerName");
+        this.logger = LoggerFactory.getLogger(loggerName);
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java:92]: nit: can remove `this.`

**Review by minwoox — APPROVED** [review]
Thanks!

**Review by ikhoon — APPROVED** [review]
Thanks! @ghkim3221

File: core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java (hunk 2 above)
Comment (ghkim3221) [reply core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java]: Thanks!

File: core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java [hunk 3]
```diff
@@ -82,6 +83,16 @@ public LoggingDecoratorBuilder logger(Logger logger) {
         return this;
     }
 
+    /**
+     * Sets the specified loggerName which is used to create a new {@link Logger}.
```
Comment (trustin) [inline core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java:87]: `Sets the name of the {@link Logger} to use when logging.` ?

**Review by trustin — COMMENTED** [review]
Thanks! Left a minor documentation comment.

File: core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java (hunk 3 above)
Comment (trustin) [reply core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java:87]: Could also add: `This method is a shortcut for `logger(LoggerFactory.getLogger(loggerName))`.

Comment (ghkim3221) [reply core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java]: Update done 😄

**Review by trustin — APPROVED** [review]
Thanks a lot, @ghkim3221 ! ❤️

**Pull request merged**